packages/manager/src/features/CloudPulse/Utils/CloudPulseWidgetUtils.test.ts

```diff
@@ -0,0 +1,129 @@
+import { formatPercentage } from 'src/utilities/statMetrics';
+
+import {
+  generateGraphData,
+  generateMaxUnit,
+  getDimensionName,
+  getLabelName,
+  mapResourceIdToName,
+} from './CloudPulseWidgetUtils';
+
+import type { CloudPulseMetricsResponse } from '@linode/api-v4';
+
+it('test generateMaxUnit method', () => {
+  const legendRowsData = [
+    {
+      data: { average: 1500, last: 1600, length: 3, max: 2000, total: 4500 },
+      format: formatPercentage,
+      legendColor: '#000',
+      legendTitle: 'linode-1',
+    },
+    {
+      data: {
+        average: 2000000,
+        last: 2100000,
+        length: 3,
+        max: 2500000,
+        total: 6000000,
+      },
+      format: formatPercentage,
+      legendColor: '#000',
+      legendTitle: 'linode-2',
+    },
+  ];
+
+  const result = generateMaxUnit(legendRowsData, 'Bytes');
```

Comment:
Overall, this is a minimalistic test method and does not cover edge cases. Please add some cases like empty array etc.,

Reply:
@ankita-akamai , is this done?

Reply:
Done, as of now, I've enhanced it, but this func is little confusing, has redundant checks for undefined cases within it even after having defined types..